cli/lib/nf3.mjs

```diff
@@ -222,14 +224,20 @@ class Nf3 {
   async estimateGas(contractAddress, unsignedTransaction) {
     let gasLimit;
     try {
-      // eslint-disable-next-line no-await-in-loop
-      gasLimit = await this.web3.eth.estimateGas({
-        from: this.ethereumAddress,
-        to: contractAddress,
-        data: unsignedTransaction,
+      const res = await axios.post(this.web3WsUrl, {
```

Comment:
This looks like a weird construct. Can you add some comments in the code so that this code is not modified later

Reply:
I can, but I don't see why it's weird. `estimateGas()` doesn't work properly on Edge, but it works fine if you call the RPC node directly. This is being addressed, but in the meantime I'm using this workaround

Reply:
hey as suggested I added a comment explaining it:
https://github.com/EYBlockchain/nightfall_3/blob/zepedro/edgefall/cli/lib/nf3.mjs#L225

that's why I resolved this comment. I'll add another comment to the `encodeABI` call, as suggested